types/type-provider.d.ts

```diff
@@ -93,3 +93,11 @@ RouteGeneric
 // review: support both async and sync return types
 // (Promise<Return> | Return | Promise<void> | void)
   : unknown
+
+/**
+ * This branded type is needed to please `typescript-eslint`.
+ *
+ * Please refer to the following Github issue for more info:
+ * https://github.com/fastify/fastify/issues/5498
+ */
+export type SafePromiseLike<T> = PromiseLike<T> & { __brand: 'SafePromiseLike' }
```

Comment:
I want to ensure we didn't use the wrong property.
It is a bit confuse to me `__brand` is not showing in the document, and `redux-tool` is using `__linterBrands`

https://github.com/fastify/fastify/issues/5498#issuecomment-2154566979

Reply:
Good point @climba03003. On the [eslint](https://typescript-eslint.io/rules/no-floating-promises/#allowforknownsafepromises) doc they recommend using `__linterBrands`:
<img width="827" alt="image" src="https://github.com/fastify/fastify/assets/3996291/ccdcd206-4159-4ae9-b663-c0034b605077">

Updated on [cb38496](https://github.com/fastify/fastify/pull/5506/commits/cb3849612b128fcd2b90da8578e2055f85366d46) ✅